actor/builtin/miner/miner_test.go

```diff
@@ -384,11 +384,58 @@ func TestMinerCommitSector(t *testing.T) {
 		// fail because commR already exists
 		res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, 4, "commitSector", nil, uint64(1), commD, commR, commRStar, th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
 		require.NoError(t, err)
-		require.EqualError(t, res.ExecutionError, "sector already committed")
+		require.EqualError(t, res.ExecutionError, "sector already committed at this ID")
 		require.Equal(t, uint8(0x23), res.Receipt.ExitCode)
 	})
 }
 
+func TestMinerSubmitPoStNextDoneChanges(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx := context.Background()
+	st, vms := th.RequireCreateStorages(ctx, t)
+
+	ancestors := th.RequireTipSetChain(t, 10)
+	origPid := th.RequireRandomPeerID(t)
+	minerAddr := th.CreateTestMiner(t, st, vms, address.TestAddress, origPid)
+	proof := th.MakeRandomPoStProofForTest()
+	done := types.EmptyIntSet()
+
+	firstCommitBlockHeight := uint64(3)
+
+	// add a sector
+	res, err := th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight, "commitSector", ancestors, uint64(1), th.MakeCommitment(), th.MakeCommitment(), th.MakeCommitment(), th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
+	require.NoError(t, err)
+	require.NoError(t, res.ExecutionError)
+	require.Equal(t, uint8(0), res.Receipt.ExitCode)
+
+	// add another sector
+	res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight+1, "commitSector", ancestors, uint64(2), th.MakeCommitment(), th.MakeCommitment(), th.MakeCommitment(), th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
+	require.NoError(t, err)
+	require.NoError(t, res.ExecutionError)
+	require.Equal(t, uint8(0), res.Receipt.ExitCode)
+
+	// finish sectors
+	done = done.Add(uint64(1))
+	done = done.Add(uint64(2))
+
+	// submit post
+	res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight+5, "submitPoSt", ancestors, []types.PoStProof{proof}, done)
+	assert.NoError(t, err)
+	assert.NoError(t, res.ExecutionError)
+	assert.Equal(t, uint8(0), res.Receipt.ExitCode)
+
+	// Check that the state NextDoneSet is updated
+	miner := state.MustGetActor(st, minerAddr)
+	storage := vms.NewStorage(minerAddr, miner)
```

Comment:
Please let me know if there is a better way to look at actor state.  This seems sloppy